### PR TITLE
Detail view for websites

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Part of #40 

#### What's this PR do?
Adds a detail view for `/api/website`

#### How should this be manually tested?
Go to `/api/website/<uuid>/` and make sure it looks right. There should be a config as well if there is a starter, and if not config should be null.

### CODE REVIEW

File: websites/views.py
Comment (mbertrand) [inline websites/views.py:29]: I think we might need 2 separate viewsets for the Website model.  One read-only viewset, accessible to unauthenticated users, currently used for listing newly published websites on the front page.  And another requiring authentication, to list and view/edit details only for the websites the user has view/edit access to (`WebsiteEditorViewSet`), that does not exclude unpublished websites?  For example, right now if I try to bring up the detail API view for an unpublished website I get a 404 response - the queryset excludes them:

```python
        queryset = Website.objects.filter(
            publish_date__lte=now_in_utc(),
        ).order_by(ordering)
```

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: websites/views.py
Comment (gsidebo) [reply websites/views.py:29]: We shouldn't need a separate viewset. We should be able to manipulate the queryset based on the request user

File: websites/serializers.py
Comment (gsidebo) [inline websites/serializers.py:53]: Is this just for convenience? I guess having `config` as a top-level property is nice to have but it seems more straightforward just to use the detail serializer for `starter` (`starter = WebsiteStarterDetailSerializer(read_only=True)`)

Comment (noisecapella) [reply websites/serializers.py:53]: That's true, I'll do that

File: websites/views.py
Comment (noisecapella) [reply websites/views.py:29]: @mbertrand Should we filter unpublished websites in this PR? I can make the change now if so

Comment (mbertrand) [reply websites/views.py:29]: Either way is fine with me (I guess it would be changed to filtering by published only for anonymous users?). If you prefer I can do that as part of the work on integrating the permission classes with the views, once this PR and the permissions PR are merged.

Comment (noisecapella) [reply websites/views.py:29]: Sure, I'll leave it to you since it might be easier to deal with the views all at once

Comment (mbertrand) [reply websites/views.py:29]: Okay, it looks like the API is already broken anyway on CI - https://github.com/mitodl/ocw-studio/issues/59

**Review by mbertrand — APPROVED** [review]

**Review by gsidebo — APPROVED** [review]
👍

**Pull request merged**
